mne/channels/montage.py

```diff
@@ -446,9 +447,11 @@ def read_dig_montage(hsp=None, hpi=None, elp=None, point_names=None,
     point_names : None | list
         If list, this corresponds to a list of point names. This must be
         specified if elp is defined.
-    unit : 'm' | 'cm' | 'mm'
-        Unit of the input file. If not 'm', coordinates will be rescaled
-        to 'm'. Default is 'mm'. This is applied only for hsp and elp files.
+    unit : 'auto' | 'm' | 'cm' | 'mm'
+        Unit of the digitizer files (hsp and elp). If not 'm', coordinates will
+        be rescaled to 'm'. Default is 'auto', which assumes 'm' for *.hsp and
+        *.elp files and 'mm' for *.txt files, corresponding to the known
+        Polhemus export formats.
```

Comment:
Thanks @agramfort ! This here would be the API change (default to `unit="auto"` instead of `unit="mm"`). It does not change the behavior for previously possible argument combination because "auto" amounts to "mm" for *.txt files and arrays.

Currently (as well as before this PR) `hsp` and `elp ` provided as arrays also get rescaled, I wonder if this could be confusing... if you agree I could add a check that raises an error or at least a warning if the head has a strange size (<10 mm or >1 m)